ToolChain/Assembler/Parser/Parser.py

```diff
@@ -20,586 +19,412 @@
+
 
 class Parser:
     """
     This class is used to parse the text format code and build a list of instruction from it.
     To do so, it has a direct link into CapuaEnvironment instruction information. Those class
     and description files are directly used in order to help build the binary code required
     to run code inside of the Capua environment. The steps are as follows:
-
     1. Verify if the file name is valid, and that it is not empty.
     2. Read the file and parse each line into a list to be evaluated one at a time.
     3. Evaluate each line and, if not an empty line, store instructions and memory references in lists.
-         - Parse each line and determine whether we're dealing with an instruction, memory reference, data, etc.
-         - If an instruction is found, determine all possible forms for this instruction (eg. ins - reg - reg)
-         - Using possible forms for an instruction, determine the correct instruction code based on given arguments.
+        - Parse each line and determine whether we're dealing with an instruction, memory reference, data, etc.
+        - If an instruction is found, determine all possible forms for this instruction (eg. InsRegReg)
+        - Using possible forms for an instruction, determine the correct instruction code based on given operands.
     4. Once all info has been evaluated, if correct, we generate the binary code for each instruction
     """
 
-    loadAddress = 0x00
-    instructionList = None
-    fileContent = None
-    referenceDict = None
-    finalSize = 0
+    relativeAddress = 0                       # Used to determine memory address of a label at a given point in the file
 
-    def __init__(self, file: str="", skipValidation=False):
+    def parse(self, text):
         """
-        The initial read of the file and parsing of the file is done when Parser is initialized.
-        This is a "one stop shop" to get the information easily and fast without requiring multiple
-        step. The point of the whole class, is, after all to get these information. No need for
-        multiple steps.
-        :param file: String, the name of the file that needs to be read in to build the code
-        :return: Nothing. Just the normal init stuff...
+        This takes in an individual line of code, parses the info and returns the fully built binary code, or returns a
+        memory reference with the appropriate offset.
+        :param text: Str, Individual line of code.
+        :return: tuple: Represents the binary instruction, the current offset, and the appropriate flag
         """
 
-        if file is None or type(file) is not str or len(file) <= 1 and not skipValidation:
-            raise ValueError("Parser error - file has to be defined")
+        self.relativeAddress = 0                # Reset offset (instruction size) for every line parsed
+        line = text.split()                     # We split the line with space as a delimiter. Each token is evaluated
```

Comment:
That is actually a good idea! It would remove the comment in one go and prevent having to manage comments at all.

Reply:
ooh very handy. Would this mean we can rename the COMMENT_FLAG to something else? Like maybe EMPTY_LINE_FLAG since the line will ultimately be empty if it only has a comment?

Reply:
That would be a decent alternative.